alica_engine/include/engine/AlicaTimer.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "engine/AlicaClock.h"
+#include "engine/IAlicaTimer.h"
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+#include <thread>
+
+namespace alicaTimer
+{
+
+class SyncStopTimerImpl : public std::enable_shared_from_this<SyncStopTimerImpl>
```

Comment:
it does not look like any additional functionality from using std::enable_shared_from_this is used here.